src/integration_fleets_tutorial_cpp.md

```diff
@@ -0,0 +1,347 @@
+
+```cpp
+
+ // Finishing tasks
+  const YAML::Node task_capabilities = rmf_fleet["task_capabilities"];
+  const std::string finishing_request_string =
+    task_capabilities["finishing_request"].as<std::string>();
+  rmf_task::ConstRequestFactoryPtr finishing_request = nullptr;
+  if (finishing_request_string == "charge")
+  {
+    finishing_request =
+      std::make_shared<rmf_task::requests::ChargeBatteryFactory>();
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is configured to perform ChargeBattery as finishing request");
+  }
+  else if (finishing_request_string == "park")
+  {
+    finishing_request =
+      std::make_shared<rmf_task::requests::ParkRobotFactory>();
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is configured to perform ParkRobot as finishing request");
+  }
+  else if (finishing_request_string == "nothing")
+  {
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Fleet is not configured to perform any finishing request");
+  }
+  else
+  {
+    RCLCPP_WARN(
+      node->get_logger(),
+      "Provided finishing request [%s] is unsupported. The valid "
+      "finishing requests are [charge, park, nothing]. The task planner will "
+      " default to [nothing].",
+      finishing_request_string.c_str());
+  }
+
+```
+
+- Adding task capabilities and performable actions to the fleet
+
+```cpp
+ // Set task planner params
+  if (!_fleet_handle->set_task_planner_params(
+      battery_system,
+      motion_sink,
+      ambient_sink,
+      tool_sink,
+      recharge_threshold,
+      recharge_soc,
+      drain_battery,
+      finishing_request))
+  {
+    RCLCPP_ERROR(
+      node->get_logger(),
+      "Failed to initialize task planner parameters");
+
+    return false;
+  }
+
+  // Currently accepting any tasks as long as they are in the config
+  const auto consider =
+    [](const nlohmann::json& description,
+      rmf_fleet_adapter::agv::FleetUpdateHandle::Confirmation& confirm)
+    {
+      confirm.accept();
+    };
+
+  const bool perform_loop = task_capabilities["loop"].as<bool>();
+  if (perform_loop)
+  {
+    _fleet_handle->consider_patrol_requests(consider);
+  }
+
+  const bool perform_delivery = task_capabilities["delivery"].as<bool>();
+  if (perform_delivery)
+  {
+    _fleet_handle->consider_delivery_requests(
+      consider, consider);
+  }
+
+  const bool perform_cleaning = task_capabilities["clean"].as<bool>();
+  if (perform_cleaning)
+  {
+    _fleet_handle->consider_cleaning_requests(consider);
+  }
+
+  // Currently accepting any actions as long as they are in the config
+  if (task_capabilities["action"])
+  {
+    std::vector<std::string> action_strings =
+      task_capabilities["action"].as<std::vector<std::string>>();
+    for (const auto& action : action_strings)
+    {
+      _fleet_handle->add_performable_action(action, consider);
+    }
+  }
+```
+
+- Calculate the Planner::StartSet for each robot using compute_plan_starts with the robot's starting pose
```

Comment:
Probably a good idea to separate explaining the fleet initializing and adding individual robot.
```suggestion
### Step 3

- Compute `Planner::StartSet` for each robot
The fleet adapter will require a `Planner::StartSet` to start planning for each robot. This object can be obtained via an `Eigen::Vector3d` of the robot's starting pose using [`rmf_traffic::agv::compute_plan_starts`](https://github.com/open-rmf/rmf_traffic/blob/main/rmf_traffic/include/rmf_traffic/agv/Planner.hpp#L916-L923):
```